test/parse.js

```diff
@@ -75,6 +75,23 @@ test('parse()', function (t) {
         st.end();
     });
 
+    t.test('allows empty arrays in obj values', function (st) {
+        st.deepEqual(qs.parse('foo[]&bar=baz', { allowEmptyArrays: true }), { foo: [], bar: 'baz' });
+        st.deepEqual(qs.parse('foo[]&bar=baz', { allowEmptyArrays: false }), { foo: [''], bar: 'baz' });
+        st.end();
+    });
+
+    t.test('should throw when allowEmptyArrays is not of type boolean', function (st) {
+        st['throws'](function () {
+            qs.parse('foo[]&bar=baz', { allowEmptyArrays: 'foobar' });
+            qs.parse('foo[]&bar=baz', { allowEmptyArrays: 0 });
+            qs.parse('foo[]&bar=baz', { allowEmptyArrays: NaN });
+            qs.parse('foo[]&bar=baz', { allowEmptyArrays: null });
+            qs.parse('foo[]&bar=baz', { allowEmptyArrays: undefined });
+        }, new TypeError('allowEmptyArrays option property can only be true or false'));
```

Comment:
each of these needs to be in its own `throws` assertion, otherwise you're not checking that all of them throw, only that one of them do.